src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml.cs

```diff
@@ -13,14 +18,145 @@ namespace NuGet.PackageManagement.UI
+            {
+                TabItem tabConsolidate = new TabItem();
+                tabConsolidate.Name = nameof(tabConsolidate);
+                tabConsolidate.Tag = "Consolidate"; //From ItemFilter Enum.
+
+                StackPanel sp = new StackPanel()
+                {
+                    Orientation = Orientation.Horizontal
+                };
+
+                TextBlock textConsolidate = new TextBlock()
+                {
+                    Name = nameof(textConsolidate),
+                    Text = Resx.Action_Consolidate
+                };
+                sp.Children.Add(textConsolidate);
+
+                SetConsolidationAutomationProperties(tabConsolidate, 0);
+
+                //The textblock that displays the count.
+                _countConsolidateContainer = new Border()
+                {
+                    Name = nameof(_countConsolidateContainer),
+                    CornerRadius = new CornerRadius(2),
+                    Margin = new Thickness(3, 0, 3, 0),
+                    Padding = new Thickness(3, 0, 3, 0),
+                    Visibility = Visibility.Collapsed,
+                    HorizontalAlignment = HorizontalAlignment.Center,
+                    VerticalAlignment = VerticalAlignment.Center
+                };
+                _countConsolidateContainer.SetResourceReference(Border.BackgroundProperty, Brushes.TabPopupBrushKey);
+                sp.Children.Add(_countConsolidateContainer);
+
+                _countConsolidate = new TextBlock()
+                {
+                    Name = nameof(_countConsolidate),
+                    HorizontalAlignment = HorizontalAlignment.Right,
+                    VerticalAlignment = VerticalAlignment.Top,
+                   // FontSize = new Binding( ="{Binding FontSize, RelativeSource={RelativeSource AncestorType=UserControl}}"
+                };
+                _countConsolidate.SetResourceReference(TextBlock.ForegroundProperty, Brushes.TabPopupTextBrushKey);
+                _countConsolidateContainer.Child = _countConsolidate;
+
+                //FontSize needs to match the UserControl.
+                Binding bindingFontSize = new Binding(nameof(TextBlock.FontSize));
+                bindingFontSize.RelativeSource = new RelativeSource(RelativeSourceMode.FindAncestor, typeof(UserControl), ancestorLevel: 1);
+                BindingOperations.SetBinding(_countConsolidate, TextBlock.FontSizeProperty, bindingFontSize);
+
+                tabConsolidate.Header = sp;
+                _tabConsolidate = tabConsolidate;
+                tabsPackageManagement.Items.Add(tabConsolidate);
+            }
+        }
+
+        private void SetConsolidationAutomationProperties(TabItem tabConsolidate, int count)
+        {
+            string automationString = null;
+            if (count > 0)
+            {
+                automationString = string.Format(
+                    CultureInfo.CurrentCulture,
+                    " {0}{1}",
```

Comment:
Why is there a leading space?

Reply:
In the equivalent code in XAML, it seems that the XAML parser can't understand "{0}", which is why we have always had a leading space there.
eg: ([from dev](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/FilterLabel.xaml#L30))
`<MultiBinding StringFormat=" {0}{1}">`

xaml.cs parses this, but I kept the space for consistency. I can test and see if it breaks. Didn't see any value in having a different format string in code-behind (which has to be maintained as a special case).

Reply:
To escape a markup extension in XAML use :
`   someProp="{}{0}{1}"`

Reply:
Changed to Rob's suggestion.